.travis.yml

```diff
@@ -12,42 +12,81 @@ addons:
+        - docker build -t openhabian/${HW}-openhabian -f Dockerfile.openhabian-x86 .
+        - docker run --name "openhabian-${HW}" --privileged -d openhabian/${HW}-openhabian;
+          docker exec -it "openhabian-${HW}" bash -c \
+            'VIRT=$(sudo virt-what); echo -e "\n\e[37mTesting openhabian install on ${VIRT} ...\n"; ./build.bash local-test && mv ~/.profile ~/.bash_profile && /etc/rc.local'
+    - env: arm32hf
       script:
-        docker run --name "bats-inst-tests" --privileged -d openhabian/rpi-openhabian
-        docker exec -it "bats-inst-tests" bash -c 'bats -r -f "installation-." .'
-    - script:
-        docker run --name "bats-dest-tests" --privileged -d openhabian/rpi-openhabian
-        docker exec -it "bats-dest-tests" bash -c 'bats -r -f "destructive-." .'
-    - stage: Shellcheck
-      script: shellcheck -x -s bash openhabian-setup.sh && echo "shellcheck openhabian-setup.sh - OK"; 
-              shellcheck -x -s bash functions/*.bash && echo "shellcheck functions/*.bash - OK"; 
-              shellcheck -x -s bash build-image/*.bash && echo "shellcheck build-image/*.bash - OK";
-              shellcheck -x -s bash build.bash && echo "shellcheck build.bash - OK"
+        - docker run --rm --privileged multiarch/qemu-user-static:register --reset
+        - export HW=rpi2
+        - sed -i 's/java_arch=64-bit$/java_arch=32-bit/' build-image/openhabian.conf
+        - docker build -t openhabian/${HW}-openhabian -f Dockerfile.openhabian-emulated .
+        - docker run --name "openhabian-${HW}" --privileged -d openhabian/${HW}-openhabian;
+          docker exec -it "openhabian-${HW}" bash -c \
+            'VIRT=$(sudo virt-what); echo -e "\n\e[37mTesting openhabian install on ${VIRT} ...\n"; ./build.bash local-test && mv ~/.profile ~/.bash_profile && /etc/rc.local'
+    - env: arm32hf
+      script:
+        - docker run --rm --privileged multiarch/qemu-user-static:register --reset
+        - export HW=rpi01
+        - sed -i 's/java_arch=64-bit$/java_arch=32-bit/' build-image/openhabian.conf
+        - docker build -t openhabian/${HW}-openhabian -f Dockerfile.openhabian-ARMv6hf-emulated .
+        - docker run --name "openhabian-${HW}" --privileged -d openhabian/${HW}-openhabian;
+          docker exec -it "openhabian-${HW}" bash -c \
+            'VIRT=$(sudo virt-what); echo -e "\n\e[37mTesting openhabian install on ${VIRT} ...\n"; ./build.bash local-test && mv ~/.profile ~/.bash_profile && /etc/rc.local'
+    - env: aarch64
+      arch: arm64
+      script:
+        - export HW=rpi3
+        - sed -i 's/java_arch=32-bit$/java_arch=64-bit/' build-image/openhabian.conf
+        - docker build -t openhabian/${HW}-openhabian -f Dockerfile.openhabian-aarch64-native .
+# Docker doesn't work on Travis ARM yet (it's still alpha), cannot access privileged FS for now ?
+        - docker run --name "openhabian-${HW}" --privileged -t -d openhabian/${HW}-openhabian;
+          docker exec -it "openhabian-${HW}" bash -c \
```

Comment:
Why use -it if the input is not an interactive TTY why not just use -i this is where Travis keeps erroring.

Reply:
Nope that only changed the error output, probably just happening because Travis support is still in Beta.

Reply:
no it was a good hint - it was the second -t on docker run
https://travis-ci.com/github/mstormi/openhabian/jobs/323103949